gramineproject/gramine
# [Pal/Linux-SGX] Correctly set masks in SIGSTRUCT

## Description of the changes 

Previously, SIGSTRUCT.ATTRIBUTEMASK (which contains the masks for enclave attributes such as DEBUG and for XFRM/XCR0 bits) and SIGSTRUCT.MISCMASK (which contains the mask for Extended SSA frame features) were incorrectly set to mirror SIGSTRUCT.ATTRIBUTES and SIGSTRUCT.MISCSELECT.

This commit replaces this incorrect logic with hard-coded values for SIGSTRUCT.ATTRIBUTEMASK and SIGSTRUCT.MISCMASK. In particular, all bits in the masks are set to one (meaning "EINIT checks each attribute"). The only exceptions currently are not-security-critical AVX and AVX512 bits -- they are not reflected in the mask, and they are set in SIGSTRUCT.ATTRIBUTES if the system supports them during generation of the `.token` file.

Useful links:
- https://github.com/openenclave/openenclave/blob/master/include/openenclave/bits/sgx/sgxtypes.h
- https://www.felixcloutier.com/x86/einit
- https://en.wikipedia.org/wiki/Control_register#XCR0_and_XSS
- "Intel SGX Explained", Sections 5.6.2 and 5.7.4, and also check Figure 77

## How to test this PR? 

All tests must pass. You can manually verify the ATTRIBUTES, MISCSELECT, and their corresponding MASKs in the outputs (e.g., when building and running the Redis example).


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gramineproject/gramine/44)
